doc/classes/TileMapLayer.xml

```diff
@@ -0,0 +1,308 @@
+	<description>
+		Node for 2D tile-based maps. A TileMapLayer uses a [TileSet] which contain a list of tiles which are used to create grid-based maps. Unlike the [TileMap] node, which is deprecated, [TileMapLayer] has only one layer of tiles. You can use several [TileMapLayer] to achieve the same result as a [TileMap] node.
+		For performance reasons, all TileMap updates are batched at the end of a frame. Notably, this means that scene tiles from a [TileSetScenesCollectionSource] may be initialized after their parent. This is only queued when inside the scene tree.
+		To force an update earlier on, call [method update_internals].
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="_tile_data_runtime_update" qualifiers="virtual">
+			<return type="void" />
+			<param index="0" name="coords" type="Vector2i" />
+			<param index="1" name="tile_data" type="TileData" />
+			<description>
+				Called with a [TileData] object about to be used internally by the TileMapLayer, allowing its modification at runtime.
+				This method is only called if [method _use_tile_data_runtime_update] is implemented and returns [code]true[/code] for the given tile [param coords].
+				[b]Warning:[/b] The [param tile_data] object's sub-resources are the same as the one in the TileSet. Modifying them might impact the whole TileSet. Instead, make sure to duplicate those resources.
+				[b]Note:[/b] If the properties of [param tile_data] object should change over time, use [method notify_runtime_tile_data_update] to notify the [TileMapLayer] it needs an update.
+			</description>
+		</method>
+		<method name="_use_tile_data_runtime_update" qualifiers="virtual">
+			<return type="bool" />
+			<param index="0" name="coords" type="Vector2i" />
+			<description>
+				Should return [code]true[/code] if the tile at coordinates [param coords] requires a runtime update.
+				[b]Warning:[/b] Make sure this function only returns [code]true[/code] when needed. Any tile processed at runtime without a need for it will imply a significant performance penalty.
+				[b]Note:[/b] If the result of this function should change, use [method notify_runtime_tile_data_update] to notify the TileMapLayer it needs an update.
+			</description>
+		</method>
+		<method name="clear">
+			<return type="void" />
+			<description>
+				Clears all cells.
+			</description>
+		</method>
+		<method name="erase_cell">
+			<return type="void" />
+			<param index="0" name="coords" type="Vector2i" />
+			<description>
+				Erases the cell at coordinates [param coords].
+			</description>
+		</method>
+		<method name="fix_invalid_tiles">
+			<return type="void" />
+			<description>
+				Clears cells that do not exist in the tileset.
+			</description>
+		</method>
+		<method name="get_cell_alternative_tile" qualifiers="const">
+			<return type="int" />
+			<param index="0" name="coords" type="Vector2i" />
+			<param index="1" name="use_proxies" type="bool" default="false" />
```

Comment:
I remember you said that proxies are meant for compatibility. Did you mean compatibility with some old system or user compatibility (i.e. refactoring I guess)?

Reply:
Both. Technically it was used when you would merge two atlases together, for compatibility with 3.x.
But it also works if you rename a source ID I believe.

Reply:
Right, but is editing source ID a likely use-case? I don't think so.
We could get rid of proxies for the TileMapLayer, to simplify the API. If that's possible.

Reply:
I'd rather keep it. I know proxies are a bit annoying in the API, but it's quite useful the when you screw up something in your TileSet. Changing an ID by mistake (or purposefully) can happen, in that case proxies can save you hours of work.

If we feel it is annoying to have this argument in the getters, then we could expose a property on the TileMapLayer itself. Something like `TileMapLayer::set_use_proxy()`.

Reply:
Changing ID by mistake is very unlikely after #79419

tbh I don't really mind the argument, other than IMO it's useless and as once discussed on chat, it can be confusing. Now is the only chance to remove it and it can be re-added later if it's needed. But I'm fine either way.

Reply:
> Changing ID by mistake is very unlikely after https://github.com/godotengine/godot/pull/79419

The thing is that people might not want to change it on purpose too, just to reorganize them. I's say it's not necessarily the best idea, but well, I'd understand why people would want this option.

Though TBH, we could probably simply only expose the "proxied" values to users, as we already have a built-in way to replace tiles by their proxied tiles in the editor. Since that's the only use case I see for this to be exposed, the problem is kind of already solved internally.